apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/controller/NotificationControllerV2.java

```diff
@@ -257,13 +259,20 @@ public void handleMessage(ReleaseMessage message, String channel) {
       logger.error("message format invalid - {}", content);
       return;
     }
+    //save deferredResults key
+    AtomicReference<String> watchKey = new AtomicReference<>();
+    deferredResults.keys().forEach(key -> {
+      if(key.equalsIgnoreCase(content)){
+        watchKey.set(key);
+      }
```

Comment:
This looks slow to me, considering there might be thousands of deferred results in the memory. 

Will it make sense to lowercase the watch keys in the first place? such as `com.ctrip.framework.apollo.configservice.util.WatchKeysUtil#assembleKey`? 

So that the keys are always lower case in deferredResults and we could lowercase the content as the search key

Reply:
最开始我也是想在assembleKey上面改，但是考虑到可能有的用户就是想要用大写字母当AppId，会导致日志打出来的信息和实际的大小写不一致，以及可能有其他的问题。